src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp

```diff
@@ -186,6 +186,11 @@ void CompileModelCacheTestBase::SetUp() {
     auto fGen = std::get<0>(funcPair);
     m_functionName = std::get<1>(funcPair);
     function = fGen(m_precision, m_batchSize);
+    // avoid auto generated friendly names
+    for (auto& iter : function->get_ordered_ops()) {
+        if (iter->get_friendly_name() == iter->get_name())
+            iter->set_friendly_name(iter->get_name() + "_test");
+    }
```

Comment:
will it help if we ignore friendly names at all?
if yes, we can fix `Model::clone()` first, then ignore friendly names in `Hash` pass

Reply:
I have reworked the solution, to minimize impact to cases, and not to modify clone() in case un-expected failures in plugins (for example in cpu, the friendly name of orginal model and cloned model needs to match in their logic).
can you a check again if any concern?

Reply:
could you please clarify why adding `_autogenerated` suffix helps here? what is the difference if we don't add it?

Reply:
hi ilya, @ilya-lavrenov , 3 solutions I tried for this bug:
1) do not clone auto generated friendly name in model->clone() (1st option we discussed offline):
    pros:  reasonable not to clone auto generated friendly name, and change is quite straight-forward
    cons: many cases are using test models without friendly name, and in cpu plugin for example, they have logic to map input/outputs using friendly name after model clone, thus we are facing quite a lot failures in multiple test components
2) completely remove the op name section (is_name_auto_generated() related) in ov::Hash pass
    pros: easy code
    cons: we are missing the name section for ops whose friendly name been set explicitly, and may change potential hash/cache results.
3) the solution in this PR, we change how we measure if the friendly name is auto generated
    pros: 
      i) previously, we define a friendly name is auto generated by comparing to get_name(), this can break if
           a) the user set friendly name happens to be same as the node.get_name()
           b) in the model->clone() case, the cloned op has different friendly name and get_name(), while this op should inherit the auto-generated characteristic from its original op.
      ii) it does not change the hash behavior, the ops whose friendly name is not auto-generated will be in the serialized xml as before, thus less chance of potential hash regressions compared to previous releases
    cons: I didn't see much side effects currently

Reply:
thanks for detailed explanations @songbell